### ISSUE TITLE
arrayBuffer error on axios with npm:specifiers

### ISSUE THREAD

**Issue by thyldrm** [denoland/deno#19341]
Hello, I tried to use the axios library with npm:specifiers. The code below returns a Buffer array for repsonse.data on the `node`.
But when I run the code in `deno` using npm:specifiers, the application closes itself as soon as the request is made.

Doesn't Axios fully work with npm:specifiers?
At the same time, how do we convert the data from the response to arrayBuffer in Deno?

```
import axios from 'npm:axios';

try {
    const zipFileUrl = 'https://api.github.com/repos/<suer>/<account>/zipball';
    const response = await axios.get(zipFileUrl, { responseType: 'arraybuffer' });
    console.log(response.data)
} catch (error) {
    console.error(error);
}
```

**Comment by thyldrm** [comment]
If we throw the same request with Deno, we can translate the incoming repos as follows.

const buffer = await response.arrayBuffer();
const unitArray = new Uint8Array(buffer);

However, the `unitArray` variable does not perform the same function as the `response.data` we obtained with the `node`.
For example, the upload for azure blobclient does not accept the `unitArray` variable. 
`blockBlobClient.uploadData(unitArray)`

**Comment by lino-levan** [comment]
I would recommend not using axios, as Deno comes with the web-standard `fetch`. In `fetch` your code would look like:

```ts
const zipFileUrl = 'https://api.github.com/repos/<suer>/<account>/zipball';
try {
  const req = await fetch(zipFileUrl);
  const res = req.arrayBuffer();
  console.log(res);
} catch (err) {
  console.error(err);
}
```
[MDN](https://developer.mozilla.org/en-US/docs/Web/API/Response/arrayBuffer)

**Comment by thyldrm** [comment]
> Ah, looks like you figured it out as I was writing my message. What azure blobclient are you using?

[@azure/storage-blob](https://www.npmjs.com/package/@azure/storage-blob)

```
const axios = require('axios');
const { BlobServiceClient } = require('@azure/storage-blob');

const connectionString = '';

const blobServiceClient = BlobServiceClient.fromConnectionString(connectionString);

const zipFileUrl = 'https://api.github.com/repos/<account>/<repo>/zipball';

const containerName = '';
const blobName = 'name.zip';


async function downloadAndUploadZip() {
    try {
      const response = await axios.get(zipFileUrl, { responseType: 'arraybuffer' });
  
      if (response.status !== 200) {
        throw new Error('error');
      }
  
      const containerClient = blobServiceClient.getContainerClient(containerName);
      const blockBlobClient = containerClient.getBlockBlobClient(blobName);
  
      await blockBlobClient.uploadData(response.data);
  
      console.log('success');
    } catch (error) {
      console.error('error', error);
    }
  }
  downloadAndUploadZip();
```

Here is a sample code that works for Node, but the same code does not work when importing packages with npm:specifiers.

**Comment by thyldrm** [comment]
First of all, when I run the code with npm:specifiers in Deno in my code, the debugger closes itself in the following line and does not print any errors.

`const response = await axios.get(zipFileUrl, { responseType: 'arraybuffer' });`

When this code is run for Node, a Buffer array is returned from response.data and the code runs and upload takes place.

I get the following error in your code
`"message": "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01//EN\"\"http://www.w3.org/TR/html4/strict.dtd\">\r \n<HTML><HEAD><TITLE>Bad Request</TITLE>\r\n<META HTTP-EQUIV=\"Content-Type\" Content=\"text/html; charset=us-ascii\"></HEAD>\r\n<BODY><h2>Bad Request - Invalid Header</h2>\r\n<hr><p>HTTP Error 400. The request has an invalid header name.</p>\r\n</BODY></HTML>\r\n"`

**Comment by lino-levan** [comment]
Weird. There's a lot of work being done on the http layer of node compatibility so maybe this'll be fixed after this lands.

**Comment by bartlomieju** [comment]
I believe we fixed this problem yesterday. I'll verify in a few hours.

**Comment by bartlomieju** [comment]
Running:
```
import axios from "npm:axios";

try {
  const zipFileUrl = "https://api.github.com/repos/denoland/deno/zipball";
  const response = await axios.get(zipFileUrl, { responseType: "arraybuffer" });
  console.log(response.data);
} catch (error) {
  console.error(error);
}
```
with Deno built from `main` branch works fine. I believe this was fixed by https://github.com/denoland/deno/pull/19327.

You can get actual `Uint8Array` by doing `console.log(response.data.buffer)`.

**Comment by bartlomieju** [comment]
I'm gonna close this issue as it's fixed by #19327.
